app/controllers/movies_controller.rb

```diff
@@ -0,0 +1,40 @@
+class MoviesController < ApplicationController
+
+  def index
+    movies = Movie.all
+
+    render json: movies.as_json( only: [:id, :title, :release_date ])
+
+  end
+
+  def create
+    movie = Movie.new(movie_params)
+    if movie.save
+      render json: { id: movie.id }, status: :ok
+    else
+      render json: {}, status: :bad_request
+    end
+  end
+
+  def show
+    movie_id = params[:id]
+    movie = Movie.find_by(id: movie_id)
+    if movie
+      render json: jsonify(movie)
+    else
+      render json: { errors: { movie_id: ["No such movie"] } }, status: :not_found
+    end
+  end
+
+
+  private
+
+  def movie_params
+    params.require(:movie).permit(:title, :overview, :release_date, :inventory)
```

Comment:
In Rails when we used forms, `params` would come back populated with a nested structure that had `movie` in it. In this API, we won't have that nested structure, so we need to take out the `.require('movie')` bit. When we do this, then most of Wave 2 starts working :)